### ISSUE TITLE
Harvesters harvest crops outside of designated zone

### ISSUE THREAD

**Issue by LachimHeigrim** [BuildCraft/BuildCraft#3487]
I'm using Harvesters on MineYourMind Regrowth server. Client contains the following BC versions: buildcraft-7.1.19.jar, buildcraft-compat-7.1.5.jar.

The Harvesters work ok, however they tend to frequently wander and harvest crops outside of their working zone. It is more frequent if the mature crops are close to border of the working zone, but I've seen one wander quite far away - from working zone, to out-of-zone point A, then even further to out-of-zone point B. Both A and B contain mature crops.

See screenshots for rough idea on the working zone setup:

![2017-01-27_17 01 04](https://cloud.githubusercontent.com/assets/9980755/22377614/f546e752-e4b2-11e6-95eb-e1ff67894b54.png)

Area is flat, there should not be any severe complications to path finding.

View from docking station towards working zone:
![2017-01-27_17 08 57](https://cloud.githubusercontent.com/assets/9980755/22377825/950c6528-e4b3-11e6-85b4-84def4472be9.png)

View from docking station towards out-of-zone point A.
![2017-01-27_17 09 05](https://cloud.githubusercontent.com/assets/9980755/22377853/a6702962-e4b3-11e6-9bc0-d4fc2e29ede9.png)

By guess it seems you are not clipping the the next path point correctly by working zone in bot's path finding algo.

I used the standard Zone Planner + Emerald Gate & Work in Zone to set the area.

**Comment by LachimHeigrim** [comment]
After regenerating the map location in Zone Planner and re-applying it on the gate (break pipe+gate, place gate, apply map location) it seems the problem is solved and bot stays in designated area.

I also simplified gate configuration from [if day -> work in area, if night -> go to station] to [if redstone signal NOT present -> work in area], not sure if that could somehow affect the bot behavior. Will examine and if problem repeats I'll update.

**Comment by AEnterprise** [comment]
how where you doing the day/night condition? robots should stay in their working areas if there is a gate action telling them that zone, if the condition on that becomes false but you do not force them back to the station they will keep working but without zone limits

**Comment by LachimHeigrim** [comment]
Emerald OR gate with light sensor, mutually exclusive conditions. Actually now I realized it is not day/night it's light levels (got confused by icons). 

Interestingly when I restored the setup, the robots are still behaving correctly. I saw the bot coming back to docking station when dark, then starting to work when it got day or I placed a torch. In no case it left the zone. This is exactly the behavior I originally expected but could not get working.

From all that I'd say I must have misconfigured the zone or have it incorrectly loaded from Zone Planner. I however repeated the same setup several times, so not sure what part was wrong. 

I suggest we close this as I can't reliably reproduce it. 

What would have helped me in this case is some way to visualizing the zone from Map Item. Since neither the item nor Zone Planner show exact world coords of polygon vertices (in my case simple rectangle) I had trial and error.
